File: 2weeks/string_sort.go
package main

import (
	"sort"
)


func solution(strings []string, n int) []string {
	sort.Slice(strings, func(i, j int) bool {
		if strings[i][n] != strings[j][n] {
			return strings[i][n] < strings[j][n]
		}
		return strings[i] < strings[j]
	})
	return strings
}
//
//func main() {
//	strings := []string{"car", "bed", "sun"}
//	solution(strings, 1)
//	for _, value := range strings {                                   test code
//		fmt.Println(value)
//	}
//}

File: 2weeks/string_sort.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
 
int i;
 
bool comp(string a, string b) {
    if (a.at(i) != b.at(i)){
        return a.at(i) < b.at(i);
    } 
    else return a < b;
}
 
vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
 
    i = n;

    sort(strings.begin(), strings.end(), comp);
    answer = strings;
 
    return answer;
}

#include <iostream>
int main() {
    vector<string> strings = {"abce", "abcd", "cdx"};
    int n = 1;

    vector <string>test = solution(strings, n);
    for (int i = 0; i < test.size(); i++){
        cout << test[i] << " ";
    }
    return 0;
}